File: epi/SpreadsheetEncoding.java
package epi;
import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;
public class SpreadsheetEncoding {
  @EpiTest(testDataFile = "spreadsheet_encoding.tsv")

  public static int ssDecodeColID(final String col) {
    //T O(N) S O(1)
    /*int rt = 0;
    if(col.isEmpty()) {
      return rt;
    }

    for(char ch : col.toCharArray()) {
      rt = rt * 26 + (ch - 'A') + 1;
    }

    return rt;*/
    int rt = epi.kt.SpreadsheetEncoding.INSTANCE.ssDecodeColID(col);
//    System.out.println(col + " " + rt + " " + epi.kt.SpreadsheetEncoding.INSTANCE.ssEncodeColID(rt));
    assert  (col.equals(epi.kt.SpreadsheetEncoding.INSTANCE.ssEncodeColID(rt)));
    return rt;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "SpreadsheetEncoding.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}

File: epi/kt/SpreadsheetEncoding.kt
package epi.kt

import java.lang.StringBuilder

object SpreadsheetEncoding {
    fun ssDecodeColID(col : String) : Int {
        var rt = 0

        for(i in col.indices) {
            rt = rt * 26 + (col[i] - 'A') + 1
        }

        return rt
    }

    fun ssEncodeColID(id : Int) : String {
        var rt = StringBuilder()
        //Could not change input para
        var vId = id

        while(vId > 0) {
            vId --;
            rt.append('A'.plus(vId % 26))
            vId /= 26
        }

        return rt.reverse().toString()
    }
}